mypy/checker.py

```diff
@@ -3949,6 +3949,8 @@ def flatten(t: Expression) -> List[Expression]:
     """Flatten a nested sequence of tuples/lists into one list of nodes."""
     if isinstance(t, TupleExpr) or isinstance(t, ListExpr):
         return [b for a in t.items for b in flatten(a)]
+    elif isinstance(t, StarExpr):
+        return flatten(t.expr)
```

Comment:
Technically this adds some non-trivial functionality. Could you please add tests that check the behavior (i.e. whether types are narrowed or nor) for situations like:
```python
isinstance(x, *(str, int))  # This should fail IIUC
isinstance(x, (int, *(str, list)))  # This should behave the same as (int, str, list)
```

Reply:
I did not understand the difference of this second test with my proposed test.

Reply:
Basically your test only tests that mypy doesn't crash, while I want:
* check that type narrowing happens when it should, and doesn't happen when it shouldn't
* check both situations where tuple appears in place and where it is referenced by a variable

Reply:
I got it, thanks.

Reply:
Done.